packages/eslint-plugin-material-ui/src/rules/rules-of-use-theme-variants.js

```diff
@@ -9,7 +9,7 @@ module.exports = {
       componentBlockNode.body.forEach((node) => {
         if (node.type === 'VariableDeclaration') {
           const propsDeclarator = node.declarations.find((declarator) => {
-            return declarator.init.name === 'props';
+            return declarator.init && declarator.init.name === 'props';
```

Comment:
@eps1lon fyi I had to add this as otherwise the lint rule was throwing.

Reply:
Thanks for the heads up :+1: 

```js
let registerEffect;
```
doesn't have an init which is why this used to throw. Don't need a test since this was just unsound access.

Reply:
Will need to close this PR, should I create a PR for this change? It hasn't thrown before, this was the first time.

Reply:
That'd be nice in case someone else stumbles over this issue. I'll take a look at type-checking these files (don't want convert to TS since we then have to figure out transpiling these rules).